### PR TITLE
mock: http server

### LINKED ISSUE #285: test: mock http server
* **Version**: 0.9.x
* **Platform**: all
* **Subsystem**: test

Setup a mock http/https server to test ShadowNode http/https client to prevent test failure on network unreachability in CI environments.

Possible related tests:

- test/node/parallel/test-http-abort.js
- test/run_pass/test_tls_connect_and_close.js

### PR DESCRIPTION
**Pull request by legendecas**
##### Checklist

- [x] `npm test` passes]

close #285

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,6 +26,8 @@ install:
 - ls node_modules/.bin
 
 script:
+- sudo node tools/http_server.js --port 80 &
+- sudo node tools/http_server.js --ssl --port 443 &
 - npm run lint
```
Comment (yorkie) [inline .travis.yml]: This step should be the head :)

File: .travis.yml [hunk 2]
```diff
@@ -13,6 +13,10 @@ cache:
   directories:
   - node_modules/
 
+addons:
+  hosts:
+  - example.com
```
Comment (yorkie) [inline .travis.yml]: Does this redirect to localhost?

File: tools/http_server.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+var http = require('http');
+var https = require('https');
+var fs = require('fs');
+var qs = require('querystring');
+var path = require('path');
+
+function handleRequest(req, res){
+  //Process Post Request
+  if(req.method === "POST"){
```
Comment (yorkie) [inline tools/http_server.js]: Single quoted :)

File: .travis.yml (hunk 2 above)
Comment (legendecas) [reply .travis.yml]: yes, it's mapped to localhost

File: tools/travis_script.py [hunk 4]
```diff
@@ -19,13 +19,8 @@ def build_iotjs(buildtype, args=[], env=[]):
...
 if __name__ == '__main__':
     test = os.getenv('OPTS')
-    if test == 'host-linux':
+    if test == 'jerryscript':
         build_jerry()
-        for buildtype in BUILDTYPES:
-            ex.check_run_cmd('./tools/build.py', [
-                '--cmake-param=-DENABLE_MODULE_ASSERT=ON',
-                '--run-test',
-                '--no-check-valgrind'])
```
Comment (yorkie) [inline tools/travis_script.py]: Why removing these builds?

Comment (legendecas) [reply tools/travis_script.py]: since `ENABLE_MODULE_ASSERT` is unused at all, it's duplicated with one another matrix.

**Pull request merged**